components/brave_sync/network_time_helper.cc

```diff
@@ -0,0 +1,58 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_sync/network_time_helper.h"
+
+#include <utility>
+
+#include "base/sequence_checker.h"
+#include "base/task/post_task.h"
+#include "components/network_time/network_time_tracker.h"
+#include "content/public/browser/browser_task_traits.h"
+#include "content/public/browser/browser_thread.h"
+
+namespace brave_sync {
+
+// static
+NetworkTimeHelper* NetworkTimeHelper::GetInstance() {
+  return base::Singleton<NetworkTimeHelper>::get();
+}
+
+NetworkTimeHelper::NetworkTimeHelper() {}
+NetworkTimeHelper::~NetworkTimeHelper() {}
+
+void NetworkTimeHelper::SetNetworkTimeTracker(
+    network_time::NetworkTimeTracker* tracker) {
+  network_time_tracker_ = tracker;
```

Comment:
you should also probably have a sequence checker DCHECK here

Reply:
fixed in https://github.com/brave/brave-core/pull/5294/commits/b2fdf8c10dd8d42e0f3add64c775cfd12b8abd57